doc/sphinx/source/glossary.rst

```diff
@@ -120,5 +120,44 @@ Glossary
    export
    
       The serialized version of a trained model that can be passed to
-      one of the nGraph backends for computation.      
+      one of the nGraph backends for computation.
+
+   NN
+
+      :abbr:`NN (Neural Network)` is an acronym for "Neural Network". NN models 
+      are used to simulate possible combinations of binary logic processing 
+      and multi-layer (multi-dimensional) paths through which a :term:`data-flow graph` 
+      may be mapped or computed. A NN does not have centralized storage; rather, 
+      a NN manifests as information stored as patterns throughout the network 
+      structure. NNs may be **Recurrent** (feedback loop) or **Nonrecurrent** 
+      (feed-forward) with regard to the network vector.
+
+   ANN
+
+      :abbr:`Artificial Neural Network (ANN)`, often abbreviated as :term:`NN`. 
+
+   RANN 
+
+      :abbr:`Recurrent Artificial Neural Network (RANN)`, often abbreviated as 
+      :term:`RNN`.
+
+
+   RNN 
+    
+      A :abbr:`Recurrent Neural Network (RNN)` is a variety of :term:`NN` where 
+      output nodes from a layer on a data-flow graph have loopback to nodes that 
+      comprise an earlier layer. Since the RNN has no "centralized" storage, this 
+      loopback is the means by which the ANN can "learn" or be trained. There are 
+      several sub-categories of RNNs. An :term:`LSTM`, for example, is a 
```

Comment:
I would remove the LSTM example in this category and replace with the traditional RNN, i.e., s_t = tanh(dot(W,x_(t-1)) + dot(U, s_(t-1)), where x is the input data and s is the output/memory.

Reply:
done